plugins/inputs/prometheus/kubernetes.go

```diff
@@ -134,76 +133,43 @@ func (p *Prometheus) watchPod(ctx context.Context, clientset *kubernetes.Clients
 	podinformer := informerfactory.Core().V1().Pods()
 	podinformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(newObj interface{}) {
-			key, err := cache.MetaNamespaceKeyFunc(newObj)
-			if err != nil {
-				p.Log.Errorf("getting key from cache %s\n", err.Error())
-			}
-
-			namespace, name, err := cache.SplitMetaNamespaceKey(key)
-			if err != nil {
-				p.Log.Errorf("splitting key into namespace and name %s\n", err.Error())
+			newPod, ok := newObj.(*corev1.Pod)
+			if !ok {
+				p.Log.Error("[BUG] Not a Pod, report it to the Github issues, please")
```

Comment:
I realize in the current situation we ignore any error, but if this error should start happening for someone, is there anything we can print to point the user or us to a root cause?

Reply:
this error shouldn't happen in practice, because we create Informer for Pods and it will be Pod objects which we receive in Add/Update/Delete handlers, so this cast will never fail. It can only start happening if future refactor of this code changes something.

Reply:
I guess I am wondering what to do if we tell someone to file a bug the only response I have is "that shouldn't happen."

I'd rather this say something like "failed to convert to a pod" and print the string representation of the interface or something to give a hint at what was trying to be converted.

Does that make sense?

Reply:
Whether it will be triggered or not is dependend entirely on the code, not on environment user runs telegraf in. I figured if they report that they see this log line in the issue, it will be trivial for devs to figure out what exactly is being passed here.


We can remove check and convert just with:

```
newPod := newObj.(*corev1.Pod)
```

which will panic should `newObj` be of the wrong type and it will have all necessary details printed

Reply:
> it will be trivial for devs to figure out what exactly is being passed here.

I would hopefuly agree, but given how hard it is to get logs from some users I'd rather we give the user as much info as possible without just telling them to file an issue.

> which will panic should newObj

While an option I'd prefer we not panic anywhere in plugins.

I think all I am after is an updated error message, like "did not receive a pod" and print the interface should something ever go wrong.